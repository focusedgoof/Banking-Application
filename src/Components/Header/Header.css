.header {
  position: sticky;
  top: 0;
  left: 0;
  right: 0;
  background-color: #092981;
  padding: 0.5% 2%;
  border-bottom: 2px solid #5cbff5;
  box-shadow: 0 2px 4px rgba(0, 0, 0, 0.1);
}

.header {
  display: flex;
  justify-content: space-between;
  align-items: center;
  z-index:1;
}

.left-header {
  margin: 0;
  display: flex;
  align-items: center;
  list-style: none;
}

.left-header .logo-container {
  display: flex;
  align-items: center;
}

.left-header .logo-container img {
  height: 40px; 
}
.left-header .logo-container p {
  margin: 5% 0;
  margin-bottom:4%;
  font-size: 2rem;
  font-weight: bold;
  color: #5cbff5;
  padding-left:5px;
  text-decoration:none !important;
}
.left-header .logo-container a {
  text-decoration: none; 
  color: #5cbff5; 
}

.right-header {
  margin: 0;
  display: flex;
  align-items: center;
  list-style-type: none;
}

.right-header li {
  margin-right: 20px;
  font-size: 18px;
  cursor: pointer; 
}

.right-header a {
  text-decoration: none;
  color: #F1F1F1;
}

.right-header a:hover {
  color: #5cbff5;
  font-weight:bold;
}

.HeaderImage {
  height: 40px;
  opacity: 0.9;  
}

.header .logout-link{
  color:#5cbff5;
  font-weight:bolder;
}

