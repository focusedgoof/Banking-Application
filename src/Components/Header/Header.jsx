import React from 'react';
import { Link, useNavigate } from 'react-router-dom';
import Logo from '../../Assets/Logo.jpg';
import PropTypes from 'prop-types';
import './Header.css';

const Header = ({ handleLogout }) => {
  const navigate = useNavigate();

  const logout = () => {
    handleLogout();
    navigate('/');
  };

  return (
    <header className="header">
        <ul className='left-header'>
          <li>
            <Link to='/' style={{textDecoration:"none"}}>
              <div className="logo-container">
                <img src={Logo} alt="logo" className='HeaderImage' />
                <p>Anna Bank</p>
              </div>
            </Link>
          </li>
        </ul>

        <ul className='right-header'>
          <li><Link to='/beneficiaries'>Beneficiaries</Link></li>
          <li><Link to='/transactions'>Transactions</Link></li>
          <li><Link to='/contact'>Contact Us</Link></li>
          <li className="logout-link" onClick={logout}>Logout</li>
        </ul>
    </header>
  );
};

Header.propTypes = {
  handleLogout: PropTypes.func.isRequired,
};

export default Header;
