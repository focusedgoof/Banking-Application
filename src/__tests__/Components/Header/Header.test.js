import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import Header from '../../../Components/Header/Header';

describe('Header Component', () => {
  it('renders header correctly', async () => {
    const handleLogout = jest.fn();
    render(
      <Router>
        <Header handleLogout={handleLogout} />
      </Router>
    );

    // Check if the logo is rendered
    expect(screen.getByAltText('logo')).toBeInTheDocument();

    // Check if the navigation links are rendered
    expect(screen.getByText('Beneficiaries')).toBeInTheDocument();
    expect(screen.getByText('Transactions')).toBeInTheDocument();
    expect(screen.getByText('Contact Us')).toBeInTheDocument();
    expect(screen.getByText('Logout')).toBeInTheDocument();
  });

  it('calls handleLogout when Logout link is clicked', async () => {
    const handleLogout = jest.fn();
    render(
      <Router>
        <Header handleLogout={handleLogout} />
      </Router>
    );

    fireEvent.click(screen.getByText('Logout'));

    // Wait for asynchronous operations to complete
    await waitFor(() => {
      expect(handleLogout).toHaveBeenCalledTimes(1);
    });
  });

  it('matches snapshot', () => {
    const handleLogout = jest.fn();
    const { asFragment } = render(
      <Router>
        <Header handleLogout={handleLogout} />
      </Router>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
