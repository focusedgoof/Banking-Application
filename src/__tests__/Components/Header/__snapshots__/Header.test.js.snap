// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="header"
  >
    <ul
      class="left-header"
    >
      <li>
        <a
          href="/"
          style="text-decoration: none;"
        >
          <div
            class="logo-container"
          >
            <img
              alt="logo"
              class="HeaderImage"
              src="Logo.jpg"
            />
            <p>
              Anna Bank
            </p>
          </div>
        </a>
      </li>
    </ul>
    <ul
      class="right-header"
    >
      <li>
        <a
          href="/beneficiaries"
        >
          Beneficiaries
        </a>
      </li>
      <li>
        <a
          href="/transactions"
        >
          Transactions
        </a>
      </li>
      <li>
        <a
          href="/contact"
        >
          Contact Us
        </a>
      </li>
      <li
        class="logout-link"
      >
        Logout
      </li>
    </ul>
  </header>
</DocumentFragment>
`;
